SwiftUIDZ/SwiftUIDZ/ProductDetailView.swift

```diff
@@ -0,0 +1,161 @@
+// ProductDetailView.swift
+// Copyright © RoadMap. All rights reserved.
+
+import SwiftUI
+
+/// Детальное описание продукта
+struct ProductDetailView: View {
+    // MARK: - Constants
+
+    private enum Constants {
+        static let yellowColorName = "pastelYellow"
+        static let purpleColorName = "pastelPurple"
+        static let darkPurpleColorName = "pastelDarkPurple"
+        static let redColorName = "pastelRed"
+        static let darkRedColorName = "pastelDarkRed"
+        static let navBarHeight: CGFloat = 30
+        static let twentyPadding: CGFloat = 20
+        static let cornerRadius: CGFloat = 8
+        static let bedImageName = "bed.double"
+        static let bedText = "Bed"
+        static let priceText = "$ 55"
+        static let heartIconImageName = "suit.heart"
+        static let buyNowButtonTitle = "Buy now"
+        static let bagIconImageName = "bag"
+        static let descriptionText = "Description"
+        static let charsText = "Chars:"
+        static let totalCharsText = "/ 150"
+    }
+
+    // MARK: - Body
+
+    var body: some View {
+        VStack {
+            GradientNavigationView(navBarHeight: Constants.navBarHeight)
+            bedImageView
+            ZStack {
+                containerShadowView
+                HStack(spacing: 70) {
+                    priceView
+                    buyButtonView
+                }
+            }
+            descriptionView
+            progressView
+            textEditorView
+            Spacer()
+        }
+        .onTapGesture {
+            hideKeyboard()
+        }
+        .navigationBarTitleDisplayMode(.inline)
+    }
+
+    // MARK: - Private Properties
+
+    @State private var descriptionText = ""
+    @State private var totalChars = 0
+    @State private var lastText = ""
+    @StateObject private var keyboardResponder = KeyboardResponder()
+
+    private var textEditorView: some View {
+        TextEditor(text: $descriptionText)
+            .border(Color.secondary.opacity(0.5), width: 1)
+            .padding(.horizontal)
+            .onChange(of: descriptionText) { text in
+                totalChars = text.count
+                if totalChars <= 150 {
+                    lastText = text
+                } else {
+                    descriptionText = lastText
+                }
```

Comment:
```suggestion
                totalChars = text.count
                if totalChars <= 150 {
                    lastText = text
                } else {
                    descriptionText = lastText
                }
```
боюсь это надо выносить во вью модель